src/rbac.rs

```diff
@@ -139,17 +140,22 @@ impl Authorization {
                         |p| p.matches(&ns),
                     );
 
-                    group_match &= m;
+                    if m {
+                        rule_match = true;
+                        break;
+                    }
                 }
 
-                if group.is_empty() {
-                    trace!(matches = true, "empty group");
+                if rule.is_empty() {
+                    rule_match = true;
+                    trace!(matches = rule_match, "empty rule");
                 } else {
-                    trace!(matches = group_match, "group");
+                    trace!(matches = rule_match, "rule");
                 }
-                rule_match &= group_match;
+                group_match &= rule_match;
```

Comment:
**suggestion:** Should this short circuit? If group match ever becomes false I think we can move on to assessing the next group because the current one is going to be false.

Reply:
Tried a quick test of this locally. Here's the short circuit logic + revised test case to exercise short circuiting.
https://github.com/ilrudie/istio-ztunnel/commit/5ec391fdd0abd410050ef5bc0167465edb422e02

Reply:
Happy to raise this as it's own PR to keep this one small and focused on the requested fix if that would be preferred.

Reply:
I included it into this one, good call